### PR TITLE
963 filter agent

### PR DESCRIPTION
**Pull request by cmclscourtney**
Filter agent with configurable query. Used for filtering visualization elements based on the data from the knowledge graph. 

See README for more details

- [x] Get config from stack to get endpoints
- [x] Provide query in volume
- [x] Document 
- [x] Include config that uses with stack

### CODE REVIEW

**Comment by cmclscourtney** [comment]
Nice to have if time

- [ ] enforce literal only filtering
- [ ] read query into builder
- [ ] if correct filters not used return sensible error
- [ ] optionally provide config to name queries and set default values
- [ ] discover Blazegraph endpoints (sim to FIA)
- [ ] walk file tree to discover queries

**Review by qhouyee — APPROVED** [review]

File: Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import jakarta.annotation.PostConstruct;
+import uk.ac.cam.cares.jps.base.query.RemoteStoreClient;
+
+@SpringBootApplication
+@RestController
+public class FilterAgentApplication {
+	BlazegraphClient blazegraphClient;
+	RemoteStoreClient remoteStoreClient;
```
Comment (gpeb2) [inline Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java:33]:
```suggestion
	private BlazegraphClient blazegraphClient;
	private RemoteStoreClient remoteStoreClient;
```

File: Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		SpringApplication.run(FilterAgentApplication.class, args);
+	}
+
+	@PostConstruct
+	private void init() {
+		blazegraphClient = BlazegraphClient.getInstance();
+	}
+
+	@GetMapping(value = "/filter")
+	public String filter(@RequestParam("subs") String substitutionString,
```
Comment (gpeb2) [inline Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java:49]:
```suggestion
	public String filter(@RequestParam("subs") Map<String, String> subsMap,
```

File: Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+			@RequestParam("query") Optional<String> queryFile, @RequestParam("namespace") Optional<String> namespace)
+			throws IOException {
+		LOGGER.info("config: {}", substitutionString);
+		Map<String, String> subsMap = OBJECT_MAPPER.readValue(substitutionString,
+				new TypeReference<Map<String, String>>() {
+				});
```
Comment (gpeb2) [inline Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java:55]:
```suggestion
```

**Review by gpeb2 — CHANGES_REQUESTED** [review]
Please test my suggestions.

File: Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java (hunk 1 above)
Comment (cmclscourtney) [reply Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java]: resolved by d957cc7

**Comment by cmclscourtney** [comment]
Make sure to remove **ALL** `-SNAPSHOT` flags

File: Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java (hunk 3 above)
Comment (cmclscourtney) [reply Agents/FilterAgent/filter-agent/src/main/java/com/cmclinnovations/filteragent/FilterAgentApplication.java]: moved inside constructor in 7a07b45

**Review by gpeb2 — APPROVED** [review]

**Pull request merged**